### ISSUE TITLE
Deploy Application Custom Resource

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#2973]
It looks like during all the refactoring of kfctl we stopped installing the application custom resource.

We should add it back because we will need it to begin defining application resources to describe Kubeflow.

/cc @kunmingg @kkasravi

**Comment by jlewi** [comment]
The package is here
https://github.com/kubeflow/kubeflow/tree/master/kubeflow/application

It looks like we need to add it to the kfctl configs so we actually install it.
https://github.com/kubeflow/kubeflow/blob/master/bootstrap/config/kfctl_default.yaml

**Comment by jlewi** [comment]
It looks like there is a bunch of jsonnet logic in that file.
https://github.com/kubeflow/kubeflow/blob/b3d30863e6c0eb716fac981d6f01c6fcb3c7ee20/kubeflow/application/application.libsonnet#L78

I think that logic is related to using jsonnet to get a list of all the components installed.

I think we can remove that logic and instead rely on the Application controller to match the application to its resources based on labels.

**Comment by jlewi** [comment]
It looks like we have a kustomize manifest for it
https://github.com/kubeflow/manifests/blob/master/metacontroller/base/crd.yaml

Should we just focus on the application resources in 0.6 and not worry to much about 0.5?

I think the kustomize package is still using our custom meta-controller implementation and we want to switch to using the controller from sig-apps.

Should we use this issue to track switching to sig-apps controller or file a new issue?

**Comment by kkasravi** [comment]
We can use this issue. I've updated #2976 with status of the application-controller.
I'm trying to get it checked in next week. (our fork of application-controller will go under components/.

**Comment by jlewi** [comment]
@kkasravi Why do we need a fork of the application controller? What changes are you making to the application controller?

Should we start by deploying the application controller and seeing how far that gets us before we start creating a fork?

**Comment by jlewi** [comment]
If we do need to maintain a fork; would it be easier logistically to manage the fork if we  create a repo in kubeflow rather than making it a sub directory?

**Comment by kkasravi** [comment]
@jlewi it would be easier to maintain a fork rather than under components. I do have a fork under my github id.

Current changes are
- [x] building with go modules
- [x] added client-gen, lister-gen, informer-gen

I anticipate additional changes.

**Comment by jlewi** [comment]
@kkasravi Why do we need those changes? What is the UX if we just use the out of box application controller?

Do we get basic monitoring?

Why not try to use the existing application controller.
See what issues it has. 
Create issues in sig-apps for the missing features.
And then discuss with sig-apps whether we should fork or try to fix upstream.

As opposed to starting by forking the controller.

**Comment by jlewi** [comment]
@kkasravi Is this done can we close this issue?

**Comment by kkasravi** [comment]
yes, there are a few application.yamls left in manifests but this issue can be closed
